x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/settings/components/edit_output_flyout/output_form_validators.tsx

```diff
@@ -369,6 +369,34 @@ export function validateDynamicKafkaTopics(value: Array<EuiComboBoxOptionOption<
         })
       );
     }
+
+    const getAllIndices = (str: string, substring: string): number[] => {
+      const indices = [];
+      let indx = str.indexOf(substring);
+      while (indx !== -1) {
+        indices.push(indx);
+        indx = str.indexOf(substring, indx + 1);
+      }
+      return indices;
+    };
+    // need to make sure that there are matching amounts of opening and closing brackets
+    const openingBracks = getAllIndices(val?.value!, '{[');
+    const closingBracks = getAllIndices(val?.value!, ']}');
+    if (openingBracks.length !== closingBracks.length) {
+      res.push(
+        i18n.translate('xpack.fleet.settings.outputForm.kafkaTopicBracketsError', {
+          defaultMessage: 'Topic should have matching amounts of opening and closing brackets',
+        })
+      );
+    }
+    // check for preceding percent sign
+    if (!openingBracks.every((item) => val?.value![item - 1] === '%')) {
+      res.push(
+        i18n.translate('xpack.fleet.settings.outputForm.kafkaTopicPercentError', {
+          defaultMessage: 'Opening brackets should be preceded by a percent sign',
+        })
+      );
+    }
```

Comment:
I think we definitely want to get this behavior under some unit tests. I know we don't have any right now for `validateDynamicKafkaTopic` but it definitely seems like an obvious case for some unit tests. Can we add a `use_output_form.test.tsx` file adjacent to this one and start off with some tests that cover this logic?